### ISSUE TITLE
Patch-to-patch fire spread

### ISSUE THREAD

**Issue by slevisconsulting** [NGEET/fates#667]
My notes from our 2020/6/19 discussion. 
Present: @lmkueppers @ckoven @jkshuman @pollybuotte @adrifoster @xuchongang @rosiealice @glemieux 

**My prep for the meeting**
1. Hierarchy reminders for my benefit
- ignitions occur at the site level (ctsm column)
- fire calculations occur at the patch level
- fire calculations depend on and affect cohort quantities, e.g. fuels and vegetation characteristics
2. McCabe & Dietze (2019) [Scaling Contagious Disturbance: A Spatially-Implicit Dynamic Model](https://www.frontiersin.org/articles/10.3389/fevo.2019.00064/full)

**Discussion during the meeting**

High level issues raised by Lara
1. Spatial scales: regional to global
Polly pointed out that many CA fires are bigger than the 4x4 km2 gridcell.
Charlie proposed and others expanded on the idea of a dynamic radius of influence both within gridcell and across gridcells.
2. Temporal scales: daily (current FATES-SPITFIRE) vs fire persistence
Lara(?) pointed out that many fires last longer than a day.
I proposed that this may be the simpler concept to tackle first.

We discussed how to determine whether a patch will burn due to spread:
- FDI calculation (0 to 1) for lightning vs setting FDI to 1 for successful ignition (from Bin’s data or from persistence/spread).
- fire_threshold concept and whether to vary this or update the FI (fire intensity) as a fire burns. The latter is very relevant in the case of crown fire (recall incomplete prior attempt to update FI due to crown fire).
- FD (fire duration) concept. We didn’t say much about this, but I understood that we will likely need to think about how this relates to fire persistence and spread.
- Charlie suggested that the adjacency matrix concept will likely reduce area burned. He explained that we can think of the current fire model as having a diagonal adjacency matrix with all ones in the diagonal. As soon as we reduce the values in the diagonal from 1 and increase the values elsewhere in the matrix from 0, the area burned will likely become less.
- The topic of determining the adjacency matrix remains unresolved, I think.

Sam to meet with Ryan (Lara, Charlie, others interested?) to discuss the technical issues of representing in FATES:
- Fire persistence beyond a day
- Fire spread across patches
- Fire spread across gridcells

Possibly useful: Is what I did before for beetles (a) relevant and (b) correct? If so, search for `mpi_allreduce` in `subroutine dynProgBB` in `/biogeochem/dynHarvestMod.F90` here:
https://github.com/ESCOMP/CTSM/compare/master...slevisconsulting:prog_beetles

**Comment by rgknox** [comment]
pinging @mdietze

**Comment by slevisconsulting** [comment]
@lmkueppers requested a list of site-level variables.

Site-level variables in the order that they appear in SFMainMod.F90 when I search for "Site":
1) acc_NI: cumulative Nesterov index over fire season
2) wind: the site-level variable is used only to calc. ROS_back
3) FDI: probability that an ignition will start a fire
4) NF: number of lighting strikes per day per km2 (now number of all ignitions)
5) frac_burnt: the site-level variable is diagnostic only

_1) More about acc_NI:_
currentSite%acc_NI is a function of t_veg24, precip24, and relhumid24. All three enter FATES as patch-level but precip and rh originate in site-level data.

fuel_moisture is a function of currentSite%acc_NI and parameters by fuel type . It is then weighted by amount of each fuel type to get currentPatch%fuel_eff_moist.

_2) More about wind:_
currentPatch%effect_wspeed = currentSite%wind * (tree_fraction * 0.4 + (grass_fraction + bare_fraction) * 0.6)  ! Recall that CTSM has its own patch-level wind, though it prob. does not translate to this effective wind speed
currentPatch%ROS_back = currentPatch%ROS_front * exp(-0.012 * currentSite%wind)

_3) More about FDI, NF, and frac_burnt:_
currentSite%FDI is a function of currentSite%acc_NI.
currentSite%NF is a function of lightning and pop_density.
currentPatch%FD = (SF_val_max_durat + 1) / (1 + SF_val_max_durat * exp(SF_val_durat_slope * currentSite%FDI))  ! So it doesn’t really vary by patch
AB = size_of_fire * currentSite%NF * currentSite%FDI  ! size_of_fire is patch level
currentPatch%frac_burnt = (min(0.99_r8, AB / km2_to_m2)) * currentPatch%area / area
currentPatch%FI = SF_val_fuel_energy * W * ROS * currentPatch%frac_burnt  ! W: tot fuel consumed by flaming front (patch)
currentSite%frac_burnt = currentSite%frac_burnt + currentPatch%frac_burnt  ! For diagnostic purposes only

_4) Other notes_
Scorch height is patch-level.
fraction_crown_burned is cohort-level.
cambial_mort is cohort-level.

**Comment by jenniferholm** [comment]
Hi all,  Can you remind me the status of this path-to-patch fire spread?  Some discussion has come up about this cross patch fire spread in different planning groups.   Do we already have cross gridcell fire spread, that mimics the cross-grid seed dispersal work?  But maybe not patch-to-patch?   Could anyone give a brief update or reminder on this?   Maybe other folks have been working on this, but we forgot to track / document it..... 

Thanks!

**Comment by ckoven** [comment]
@glemieux and I went down this route a while ago, but did not end up with a completed product. I will have to try to remember if there were usable developments that one could build off of.

**Comment by glemieux** [comment]
@jenniferholm I don't think we have any work on gridcell-to-gridcell spread for fates specifically at this point, but @adrifoster or @samsrabin might know more.  

The patch-to-patch work stopped at the conceptual planning phase as there were debates on how to model this in the CA fates call.  Here's one of the presentations Charlie and I gave on our proposal: https://docs.google.com/presentation/d/15wEGNxMNRO5tgkoCv8cAAd92kVZlJsFIWrJk1f_GTIU/edit?slide=id.g1182a8999fa_0_7#slide=id.g1182a8999fa_0_7

**Comment by jenniferholm** [comment]
Thanks Charlie and Greg for the reminder!  I thought there already was a prototype, working version of this patch-to-patch spread, but was mistaken.   
Greg - I do remember those conceptual slides on the matrix of self-adjacency.  I guess let's wait and see if this subgrid wildfire spread between patches is a needed task for either NGEE 2.0 or E3SM, or CA NASA studies and then we can go from there to see if there are resources.   Thanks!
